### PR TITLE
[ISSUE-811] Remove annotation in drive metadata for deleted volume

### LINKED ISSUE #811: Removing the drive volume annotation if volume has been deleted
**Description**
Volume annotation is not deleted from a related drive annotations, if a volume has been deleted. So, it may have cases like:
`root@servicenode ~]# oc describe drive bd1b4aa8-995f-4ef9-b909-7ff6ddd1f707
Name:         bd1b4aa8-995f-4ef9-b909-7ff6ddd1f707
Namespace:    
Labels:       app=csi-baremetal
              app.kubernetes.io/name=csi-baremetal
Annotations:  status/pvc-04ef52dd-2f00-46c8-96c6-58afbcf92555: REMOVED
              status/pvc-068603a9-682c-48b8-b892-7d9125545f6e: REMOVED
              status/pvc-06ef307f-755f-4616-93e6-5b189e68b8cc: REMOVED
              status/pvc-08108a32-c439-4f72-b8da-f1a2e2fde18e: REMOVED
...
API Version:  csi-baremetal.dell.com/v1
Kind:         Drive
Metadata:
  Creation Timestamp:  2021-12-27T10:52:59Z
  Generation:          145
  Managed Fields:
    API Version:  csi-baremetal.dell.com/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:status/pvc-04ef52dd-2f00-46c8-96c6-58afbcf92555:
          f:status/pvc-068603a9-682c-48b8-b892-7d9125545f6e:
          f:status/pvc-06ef307f-755f-4616-93e6-5b189e68b8cc:
...
` 

**Environment:**
 - Version: 1.1.0

**Steps to Reproduce**
1. Apply and delete any application to have volumes CREATED:
`root@ubuntu:/home/emc# kubectl get volumes
NAME                                       SIZE        STORAGE CLASS   HEALTH   CSI_STATUS   LOCATION                               NODE
pvc-317e32c6-863e-44cd-91fb-2d2c46d2920d   105906176   HDD             GOOD     CREATED      1557cc63-fc1a-4aba-8704-953eb59ed46c   486cc4c9-6818-4e06-b2b0-730a1e4b34b7
pvc-4176ad74-c79c-40fa-9018-f7489dccce0d   105906176   HDD             GOOD     CREATED      9989f170-7d61-49a1-8692-cec7ae6125fa   8e27c561-0e0c-45f8-86a6-45a2f38df91f
...
`
2. Delete volumes:
`root@ubuntu:/home/emc# kubectl delete volumes pvc-317e32c6-863e-44cd-91fb-2d2c46d2920d pvc-4176ad74-c79c-40fa-9018-f7489dccce0d...`
 
**Expected behavior**
A drive annotations shouldn't have deleted pvc annotations.

### PR DESCRIPTION
**Pull request by mitsaa**
### Resolves #811 

_Remove annotation in drive metadata for deleted volume_

## PR checklist
- [X] Add link to the issue
- [X] Choose Project
- [X] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
1. Deploy CSI
2. Deploy an application 
3. Delete an application
4. Delete all pvc-es
5. Make sure that a drive doesn't have a "status/pvc-name" in its metadata

### CODE REVIEW

File: pkg/node/volumemgr.go [hunk 1]
```diff
@@ -301,7 +301,8 @@ func (m *VolumeManager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 			// drive must be present in the system
 			drive, _ := m.crHelper.GetDriveCRByVolume(volume)
 			if drive != nil {
-				m.addVolumeStatusAnnotation(drive, volume.Name, apiV1.Removed)
+				delete(drive.Annotations,
```
Comment (rkiselenko) [inline pkg/node/volumemgr.go:304]: It's better to use special methods for annotations:

```golang
annotations := obj.GetAnnotations()
delete(annotations, annotation)
obj.SetAnnotations(annotations)
// Update obj
```

Comment (rkiselenko) [inline pkg/node/volumemgr.go:304]: It's better to use special methods for annotations:

```golang
withAnnotation := obj.DeepCopyObject()
annotations := obj.GetAnnotations()
delete(annotations, annotation)
obj.SetAnnotations(annotations)
// Update obj
```

**Review by rkiselenko — CHANGES_REQUESTED** [review]

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: pkg/node/volumemgr.go [hunk 2]
```diff
@@ -301,7 +301,8 @@ func (m *VolumeManager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 			// drive must be present in the system
 			drive, _ := m.crHelper.GetDriveCRByVolume(volume)
 			if drive != nil {
-				m.addVolumeStatusAnnotation(drive, volume.Name, apiV1.Removed)
```
Comment (mishoyama) [inline pkg/node/volumemgr.go:304]: What was the goal for adding annotation? If I remember correctly the idea is to update drive CR and trigger reconcile loop to do extra work...

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: pkg/node/volumemgr.go (hunk 1 above)
Comment (mitsaa) [reply pkg/node/volumemgr.go:304]: func (meta *ObjectMeta) GetAnnotations() map[string]string            { return meta.Annotations }
(meta *ObjectMeta) SetAnnotations(annotations map[string]string) { meta.Annotations = annotations }
=)
But I should to agree that your suggestion is more elegant.

File: pkg/node/volumemgr.go (hunk 2 above)
Comment (safronovD) [reply pkg/node/volumemgr.go:304]: Removing annotation will trigger reconcilation too

**Review by rkiselenko — APPROVED** [review]

**Review by safronovD — APPROVED** [review]

**Review by mishoyama — APPROVED** [review]

**Pull request merged**